src/test/run-make/crate-hash-rustc-version/Makefile

```
@@ -1,23 +1,28 @@
 -include ../../run-make-fulldeps/tools.mk
 
-# only-linux
-
-# This test only works on Linux, since it uses lib*.so files.
-
 FLAGS := -Cprefer-dynamic -Zsymbol-mangling-version=v0
+UNAME := $(shell uname)
+ifeq ($(UNAME),Linux)
+  EXT=".so"
+  NM_CMD="nm -d"
+endif
+ifeq ($(UNAME), Darwin)
+  EXT=".dylib"
+  NM_CMD="nm"
+endif
 
 all:
 	# a.rs is a dylib
-	$(RUSTC) a.rs --crate-type=dylib -o $(TMPDIR)/liba.so ${FLAGS}
+	$(RUSTC) a.rs --crate-type=dylib -o $(TMPDIR)/liba$(EXT) $(FLAGS)
 	# Write symbols to disk.
-	nm -D $(TMPDIR)/liba.so > $(TMPDIR)/symbolsbefore
+	$(NM_CMD) $(TMPDIR)/liba$(EXT) > $(TMPDIR)/symbolsbefore
```

```suggestion
	$(NM_CMD) $(call DYLIB,a) > $(TMPDIR)/symbolsbefore
```
should work I think.